core/templates/dev/head/pages/exploration_player/AudioTranslationManagerService.js

```diff
@@ -0,0 +1,134 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Service to manage the current language being
+ * used for audio translations as well as related modals.
+ */
+
+oppia.factory('AudioTranslationManagerService', [
+  '$modal', 'ExplorationPlayerStateService', 'AudioPlayerService',
+  function(
+      $modal, ExplorationPlayerStateService, AudioPlayerService) {
+    var _currentAudioLanguageCode = null;
+    var _allLanguageCodesInExploration = null;
+
+    var _init = function() {
+      _allLanguageCodesInExploration =
+        ExplorationPlayerStateService.getAllAudioLanguageCodes();
+
+      // TODO(tjiang11): Define an order. Preferably, we'd want
+      // to promote the user's preferred languages, and promote
+      // the exploration's default language if available.
+      _allLanguageCodesInExploration.sort();
+
+      if (_allLanguageCodesInExploration.length == 1) {
+        _currentAudioLanguageCode = _allLanguageCodesInExploration[0];
+      }
+
+      if (_allLanguageCodesInExploration.length > 1) {
+        // TODO(tjiang11): Need to use a pick-language modal instead of
+        // defaulting to a language.
+        _currentAudioLanguageCode = _allLanguageCodesInExploration[0];
+      }
+    };
+
+    var _showAudioTranslationSettingsModal = function() {
+      $modal.open({
+        templateUrl: 'modals/audioTranslationSettings',
+        resolve: {},
+        controller: [
+          '$scope', '$filter', '$modalInstance',
+          'AudioTranslationManagerService',
+          function(
+              $scope, $filter, $modalInstance,
+              AudioTranslationManagerService) {
+            $scope.selectedLanguage =
+              $filter('languageDescription')(_currentAudioLanguageCode);
+            $scope.allLanguageCodes =
+              AudioTranslationManagerService
+                .getAllLanguageCodesInExploration();
+
+            $scope.save = function() {
+              var selectedLanguageCode = 
+                $filter('languageCode')($scope.selectedLanguage);
+              $modalInstance.close({
+                languageCode: selectedLanguageCode
+              });
+            };
+          }
+        ]
+      }).result.then(function(result) {
+        if (_currentAudioLanguageCode !== result.languageCode) {
+          _currentAudioLanguageCode = result.languageCode;
+          AudioPlayerService.stop();
+          AudioPlayerService.clear();
+        }
+      });
+    };
+
+    return {
+      init: function() {
+        _init();
+      },
+      getCurrentAudioLanguageCode: function() {
+        return _currentAudioLanguageCode;
+      },
+      getAllLanguageCodesInExploration: function() {
+        return _allLanguageCodesInExploration;
+      },
+      showAudioTranslationSettingsModal: function() {
+        return _showAudioTranslationSettingsModal();
+      }
+    };
+  }]);
+
+oppia.filter('languageDescriptions', ['$filter', function($filter) {
```

Comment:
Try to replace these filters with service functions if possible. I think you don't actually need any of them.